beets/ui/__init__.py

```diff
@@ -1093,43 +1103,158 @@ def _open_library(config):
     return lib
 
 
-def _raw_main(args, lib=None):
+class BeetsCLI(click.MultiCommand):
+
+    def _commands(self, ctx):
+        # FIXME: Awful performance, cache per ctx
```

Comment:
Not really, we already get the ctx object passed.

Perhaps the cache should be stored on the ctx object itself though.